docs/data/data-grid/column-dimensions/column-dimensions.md

```diff
@@ -58,6 +58,43 @@ To capture changes in the width of a column there are two callbacks that are cal
 - `onColumnResize`: Called while a column is being resized.
 - `onColumnWidthChange`: Called after the width of a column is changed, but not during resizing.
 
+## Autosizing [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
+
+`DataGridPro` also allows for autosizing the columns' dimensions based on their content. It can be autosized by 3 methods:
+
+- Adding the `autosizeOnMount` prop,
+- Double-clicking a column header separator on the grid,
+- Calling the `apiRef.current.autosizeColumns(options)` API method.
+
+You can pass options directly to the API method when you call it. If you need to configure autosize for the other two methods, you can provide the options in the `autosizeOptions` prop. Note that for the separator double-click method, the clicked column will be passed to the `.columns` option, overriding the value of if it exists in `autosizeOptions`. In all cases, the `.minWidth` and `.maxWidth` options of columns will always be respected.
+
+```tsx
+<DataGridPro
+  {...otherProps}
+  autosizeOptions={{
+    columns: ['name', 'status', 'createdBy'],
+    includeOutliers: true,
+    includeHeaders: false,
+  }}
+/>
+```
+
+{{"demo": "ColumnAutosizing.js", "disableAd": true, "bg": "inline"}}
+
+:::warning
+Autosizing has no effect if dynamic row height is enabled.
+:::
+
+:::warning
+The datagrid can only autosize based on the currently rendered cells.
```

Comment:
It should be at minimum two words to be coherent with `DataGrid` and https://www.notion.so/mui-org/Style-conventions-for-MUI-components-and-products-dff7d2d7d1ba4a4abd47f48cf345b800?pvs=4#8f86b8e3c0fb49e1b4df3276efb3848e.

```suggestion
The data grid can only autosize based on the currently rendered cells.
```

x the other cases.

However, I suspect Sam would push for as it references our data grid, not data grids in general.

```suggestion
The Data Grid can only autosize based on the currently rendered cells.
```

but this hasn't been applied very consistently so far, so unclear.

Reply:
I'll switch to `data grid` for now, but just to note I don't love `data grid` and even less `Data Grid` because it can be confused for a variant of [Grid](https://mui.com/material-ui/react-grid/).

Reply:
To illustrate the move towards Data Grid: https://github.com/mui/mui-x/pull/5781/files, which is equivalent to the normalization ongoing in the core by @samuelsycamore.

History, grid would intuitively mean data gird, e.g. https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/data-grids/ uses this a bit interchangeably.

Reply:
I would strongly prefer to see Data Grid capitalized to emphasize that we're talking about the product, not the general concept. `DataGrid` is an outdated convention that we're slowly trying to phase out of the docs. If you need to make it clear that you're talking about the code specifically, use `<DataGrid />`. Overall it's less what important _what_ we use so long as we apply it _consistently_.